### PR TITLE
Bump protocol version to 18

### LINKED ISSUE #3202: Bump protocol version to 18
We need to bump the `CURRENT_LEDGER_PROTOCOL_VERSION` to 18 before the v18.0.0 release.

### PR DESCRIPTION
**Pull request by jonjove**
# Description

Resolves #3202 

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/herder/test/UpgradesTests.cpp [hunk 1]
```diff
@@ -2346,7 +2357,12 @@ TEST_CASE("upgrade from cpp14 serialized data", "[upgrades]")
 TEST_CASE("upgrade flags", "[upgrades][liquiditypool]")
 {
     VirtualClock clock;
-    auto app = createTestApplication(clock, getTestConfig());
+    // Do our setup in version 1 so that for_versions_* below do not
```
Comment (jonjove) [inline src/herder/test/UpgradesTests.cpp:2360]: Nitpick: genesis version is 0, so all of these 1s should be 0s.

File: src/herder/test/UpgradesTests.cpp [hunk 2]
```diff
@@ -2219,17 +2219,28 @@ TEST_CASE("simulate upgrades", "[herder][upgrades][acceptance]")
 TEST_CASE("upgrade invalid during ledger close", "[upgrades]")
 {
     VirtualClock clock;
-    auto app = createTestApplication(clock, getTestConfig());
+    // Do our setup in version 1 so that for_versions_* below do not
```
Comment (jonjove) [inline src/herder/test/UpgradesTests.cpp:2222]: Nitpick: genesis version is 0, so all of these 1s should be 0s.

File: src/transactions/test/SetTrustLineFlagsTests.cpp [hunk 3]
```diff
@@ -443,8 +443,8 @@ TEST_CASE("revoke from pool",
         }
     };
 
-    auto depositIntoPool = [&](TestAccount& account, Asset const& assetA,
-                               Asset const& assetB) {
+    auto depositIntoPool1 = [&](TestAccount& account, Asset const& assetA,
```
Comment (jonjove) [inline src/transactions/test/SetTrustLineFlagsTests.cpp:446]: Can we give these more descriptive names? 1 and 2 doesn't really tell me anything about what they do/why they're different.

**Comment by MonsieurNicolas** [comment]
r+ 2138c54c1d5996a405101d87173a19544078f9e2

**Pull request merged**